src/pages/ProductList/index.js

```diff
@@ -16,17 +16,34 @@ import { ProductsWrap } from './styled'
 
 const Products = ({ match, location, addProduct }) => {
   const { page } = qs.parse(location.search.substr(1))
+  const [count, setCount] = useState(25)
 
   const { data: res, isLoading } = useApi(
-    () => getProducts({ page: { number: page } }),
+    () => getProducts({ page: { number: page, size: count } }),
     [page]
   )
 
+  useEffect(() => {
+    // TODO: Rerender here
+    // const res = useApi(
+    //   () => getProducts({ page: { number: page, size: count } }),
+    //   [page]
+    // )
+    // ERROR: Hooks can only be called inside the body of a function component.
+  })
+
   const handleAddToCart = productId => addProduct(productId)
+  const handleCount = event => setCount(event.target.value)
 
   return (
     <Layout>
       <H1 textAlign="center">E-Commerce app</H1>
+      Products per page:
+      <select value={count} onChange={handleCount} onBlur={handleCount}>
+        <option value="9">9</option>
```

Comment:
Using both `onChange` and `onBlur` is the right way, however, you also want to check that the value in `event.target.value` is different from `count`, and call API only if it's different.